plugins/ingest/src/main/java/org/elasticsearch/plugin/ingest/PipelineStore.java

```diff
@@ -69,6 +91,7 @@ public PipelineStore(Settings settings, ThreadPool threadPool, Environment envir
 
     @Override
     protected void doStart() {
+        client = injector.getInstance(Client.class);
```

Comment:
I wish there was a better way... if the `Client` gets injected either via setter or constructor then there is an cyclic dependency error. 

(the node client can't be created until all transport services (transport actions, action filters etc.) have been created. The IngestActionFilter relies indirectly on PipelineStore which has Client as a dependency. )